app/src/androidTest/java/com/github/sdp_begreen/begreen/dialog/ContestMapDialogTest.kt

```diff
@@ -147,23 +164,27 @@ class ContestMapDialogTest {
                 }
             }
 
+            withContext(Dispatchers.IO) {
+                // Add some arbitrary delay to let time to the map to initialize
+                Thread.sleep(MAP_INITIALIZATION_DELAY)
+            }
```

Comment:
I was not a big fan of always waiting those 10s before the test and the one below and I think I found a better way of waiting for the map to be ready:
First, we can add the following line in `ContestMapDialog`(at line 95, in the function `onMapReady`, just below `this.map = map`):
`map.setContentDescription("MAP READY")`.
Then, we could replace the 3 lines above by:

```suggestion
            assertThat(
                device.wait(Until.hasObject(By.desc("MAP READY")), MAP_INITIALIZATION_DELA),
                `is`(true)
            )
```
That way, we instantly stop waiting when the map is ready. We can also apply this to the test below, and we could maybe rename the parameter `MAP_INITIALIZATION_DELAY` to something more relevant such as `TIMEOUT` or something like that. Let me know what you think about it!

Reply:
I will give it a try! Thanks a lot for your investigation, hopefully it works. I also hated putting arbitrary delay in the code, but couldn't come with something better that works on the CI.

I'll let you know if it works :)

Reply:
It works really well, thank a lot for this ! :smile: